### ISSUE TITLE
Swap `DexIncentive` with `DexPremium`, making `DexPremium` reward 'DexCurrencyId` (DRAM/MENA)

### ISSUE THREAD

**Issue by balqaasem** [setheum/setheum#365]
Swap `DexIncentive` with `DexPremium`, making `DexPremium` reward 'DexCurrencyId` (DRAM/MENA). And rewarding the DexInflation into the `DexPremium`

For example:
```
fn dex_inflation(amount: Balance) -> Self::Balance {
	let dexer_currency_id = T::DexCurrencyId::get();
	let total_issuance = T::MultiCurrency::total_issuance(dexer_currency_id),
	let max_supply = T::SerpTreasury::dexer_max_supply();
	ensure!(amount + total_issuance <= max_supply;

	T::SerpTreasury::issue_dexer(&amount);
	Ok(())
}
```

**Comment by balqaasem** [comment]
Better to pre-mine it and then just keep depositing through the Financial council as the Premium Incentives

**Comment by balqaasem** [comment]
Or

> Better to pre-mine it and then just keep depositing through the Financial council as the Premium Incentives

Or just issue it  in certain amounts (`standard_inflation_rate`) until it ends at `MaxSupply` in about 2 yrs.

**Comment by balqaasem** [comment]
The `MaxSupply` will be set in the `SerpTreasury` for only `DexCurrencyId`.

`MaxSupply` for the Currencies:
- DRAM = 1.032 Billion (1,032,000,000); (4X the `InitialSupply` of the DNAR);

- MENA = 10.32 Billion (10,320,000,000); (10X that of the DRAM) (4X the `InitialSupply` of the NEOM);

Ref: Set MaxSupply for DexCurrencyId in all networks #369

**Comment by balqaasem** [comment]
> The `MaxSupply` will be set in the `SerpTreasury` for only `DexCurrencyId`.
> 
> `MaxSupply` for the Currencies:
> 
> * DRAM = 1.032 Billion (1,032,000,000); (4X the `InitialSupply` of the DNAR);
> * MENA = 10.32 Billion (10,320,000,000); (10X that of the DRAM) (4X the `InitialSupply` of the NEOM);
> 
> Ref: Set MaxSupply for DexCurrencyId in all networks #369

So, ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ this is the updated denomination.

**Comment by balqaasem** [comment]
I suggest we distribute in a time of about 20 years... That is enough, not too small and not too large, will be more bubble-resistant.

**Comment by balqaasem** [comment]
> I suggest we distribute in a time of about 20 years... That is enough, not too small and not too large, will be more bubble-resistant.

One Generation is good enough!

**Comment by balqaasem** [comment]
Closes in #389

**Comment by balqaasem** [comment]
Support DexPremium.
Closes in #389

**Comment by balqaasem** [comment]
Changed to GovernanceCurrencyId, and as staking rewards provided alongside setters too as staking rewards
